cosmo_tester/test_suites/image_based_tests/component_test.py

```diff
@@ -0,0 +1,170 @@
+import retrying
+
+from cosmo_tester.framework import util
+from cosmo_tester.framework.examples import get_example_deployment
+
+
+def _infra(image_based_manager, ssh_key, logger, tenant, test_config):
+    example = get_example_deployment(
+        image_based_manager, ssh_key, logger, tenant, test_config,
+        upload_plugin=False)
+
+    example.blueprint_file = util.get_resource_path(
+        'blueprints/component/fake_infra.yaml'
+    )
+    example.blueprint_id = 'infra'
+    example.deployment_id = 'infra'
+    example.inputs = {}
+    return example
+
+
+def _app(image_based_manager, ssh_key, logger, tenant, test_config):
+    example = get_example_deployment(
+        image_based_manager, ssh_key, logger, tenant, test_config,
+        upload_plugin=False)
+
+    example.blueprint_file = util.get_resource_path(
+        'blueprints/component/component.yaml'
+    )
+    example.blueprint_id = 'app'
+    example.deployment_id = 'app'
+    example.inputs['agent_user'] = image_based_manager.username
+    example.create_secret = False   # don't try to create it twice
+    return example
+
+
+def _nesting_app(image_based_manager, ssh_key, logger, tenant, test_config):
+    example = get_example_deployment(
+        image_based_manager, ssh_key, logger, tenant, test_config,
+        upload_plugin=False)
+
+    example.blueprint_file = util.get_resource_path(
+        'blueprints/component/nesting_component.yaml'
+    )
+    example.blueprint_id = 'nesting_app'
+    example.deployment_id = 'nesting_app'
+    example.inputs['agent_user'] = image_based_manager.username
+    example.create_secret = False   # don't try to create it twice
+    return example
+
+
+def test_component(image_based_manager, ssh_key, logger, test_config):
+    tenant = 'test_component'
+    infra = _infra(image_based_manager, ssh_key, logger, tenant, test_config)
+    app = _app(image_based_manager, ssh_key, logger, tenant, test_config)
+
+    # We're uploading a blueprint that creates an infrastructure for a VM,
+    # and then exposes capabilities, which will be used in the application
+    logger.info('Uploading infrastructure blueprint.')
+    infra.upload_blueprint()
+    logger.info('Deploying parent component blueprint.')
+    app.upload_and_verify_install()
+
+    with util.set_client_tenant(app.manager.client, tenant):
+        # verify both deployments are installed
+        deployments = app.manager.client.deployments.list()
+        assert len(deployments) == 2
+
+        nodes = app.manager.client.nodes.list()
+        assert len(set(node.deployment_id for node in nodes)) == 2
+        for node in nodes:
+            if node.type == 'cloudify.nodes.Compute':
+                assert node.number_of_instances == 1
+
+    logger.info('Testing custom execute operation.')
+    util.run_blocking_execution(
+        app.manager.client, 'app', 'execute_operation', logger,
+        params={'operation': 'maintenance.poll', 'node_ids': 'app'},
+        tenant=tenant)
```

Comment:
Should this operation cause something you can assert on (e.g. a runtime property)?